src/main/java/XDirectionProject/service/impl/LifterServiceImpl.java

```diff
@@ -25,13 +27,42 @@ public Lifter addLifter(Lifter lifter) {
     @Override
     public Lifter findLifterByEmail(String email) {
         return lifterRepository.findLifterByEmail(email)
-                .orElseThrow(() -> new NonExistingEntityException(NON_EXISTING_ENTITY));
+                .orElseThrow(() -> new NonExistingEntityException(Constants.NON_EXISTING_ENTITY));
     }
 
     @Override
     public void deleteLifter(String email) {
         Lifter lifterToBeDeleted = lifterRepository.findLifterByEmail(email)
-                .orElseThrow(() -> new NonExistingEntityException(NON_EXISTING_ENTITY));
+                .orElseThrow(() -> new NonExistingEntityException(Constants.NON_EXISTING_ENTITY));
         lifterRepository.delete(lifterToBeDeleted);
     }
+
+    @Override
+    public Lifter updateLifter(String email, LifterDTO lifterDTO) {
+        Lifter lifterToBeUpdated = lifterRepository.findLifterByEmail(email)
+                .orElseThrow(() -> new NonExistingEntityException(Constants.NON_EXISTING_ENTITY));
+        Lifter updatedLifter = updateLifterDetails(lifterToBeUpdated, lifterDTO);
+
+        return lifterRepository.save(updatedLifter);
+
+    }
+
+    private Lifter updateLifterDetails(Lifter lifter, LifterDTO updateDetails) {
+        if (Objects.nonNull(updateDetails.getFirstName()))
+            lifter.setFirstName(updateDetails.getFirstName());
+        if (Objects.nonNull(updateDetails.getLastName()))
+            lifter.setLastName(updateDetails.getLastName());
+        if (Objects.nonNull(updateDetails.getEmail())) {
+            if (!lifterRepository.existsLifterByEmail(updateDetails.getEmail()))
+                lifter.setEmail(updateDetails.getEmail());
+        }
+        if (updateDetails.getAge() != 0)
```

Comment:
The point of the if and else statements is to check whether there is some value to search about. That's why we can't validate them because if they are not valid they won't pass, but in our case, it will create an SQL query without any criteria for searching.

Reply:
Fine, but lets consider validating them where appropriate.

Reply:
I think there should be validation in the DTO.I planned it to be added in one of the next development cycles.